packages/xpetra/sup/Utils/Xpetra_TripleMatrixMultiply.hpp

```diff
@@ -147,6 +147,30 @@ namespace Xpetra {
                         fillParams);
       }
 
+      // transfer striding information
+      RCP<const Map> domainMap = Teuchos::null;
+      RCP<const Map> rangeMap  = Teuchos::null;
+
+      const std::string viewLabel("stridedMaps");
+      const size_t        blkSize = 1;
+      std::vector<size_t> stridingInfo(1, blkSize);
+      LocalOrdinal        stridedBlockId = -1;
+      
+      if (R->IsView(viewLabel)) {
+        rangeMap  = transposeR ? R.getColMap(viewLabel) : R.getRowMap(viewLabel);
+      } else {
+        rangeMap  = transposeR ? R.getDomainMap()       : R.getRangeMap();
+        rangeMap  = Xpetra::StridedMapFactory<LocalOrdinal, GlobalOrdinal, Node>::Build(rangeMap,  stridingInfo, stridedBlockId);
+      }
+      
+      if (P->IsView(viewLabel)) {
+          domainMap = transposeP ? P->getRowMap(viewLabel) : P->getColMap(viewLabel);
+      } else {
+        domainMap = transposeP ? P->getRangeMap()        : P->getDomainMap();
+        domainMap = Xpetra::StridedMapFactory<LocalOrdinal, GlobalOrdinal, Node>::Build(domainMap, stridingInfo, stridedBlockId);
```

Comment:
Do you need to specify template arguments here? Or can you just write `domain_map = StridedMapFactory::Build(...);`?

Reply:
Yes, that's correct. The logic is adapted from `Xpetra_MatrixMatrix`